nodeadm/internal/configprovider/userdata.go

```diff
@@ -85,6 +109,14 @@ func parseMultipart(userDataReader *multipart.Reader) (*internalapi.NodeConfig,
 				if err != nil {
 					return nil, err
 				}
+				nodeConfigPart, err = decodeIfBase64(nodeConfigPart)
+				if err != nil {
+					return nil, err
+				}
+				nodeConfigPart, err = decompressIfGZIP(nodeConfigPart)
+				if err != nil {
+					return nil, err
```

Comment:
nit: theres a custom error message around the calls above but not here :(

Reply:
actually the more i think on this, should we be more tolerant and `break` on these errors so that we can try the remaining parts?